### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Osoro254Alex**
> ## I did the following in this Milestone
> * Used Postgres as database.
> * Used devise for authentication.
> * Validated all user input to make sure that anyone with bad intentions cannot compromise the app.
> * The project is deployed and accessible [online](https://osoro.onrender.com).
> * ## :link: Presentation
> * [Presentation Video](https://www.loom.com/share/61b84b8549fc4796ab774a1b20696951)
> * Created unit and integration tests for all the most important components.
> * Database schema reflects the following structure:
> 
>image
> 
> #### Interactions
> * Splash screen
>   
>   * A simple page with the name of your app (yes, you need to choose one), and links to the sign-up and log-in pages.
> * Sign up and login pages
>   
>   * The user should be able to register in the app with full name, email, and password (all mandatory).
>   * The user can log into the app using email and password.
>   * If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
> * Home page (categories page)
>   
>   * When the user logs in, they are presented with the categories page.
>   * For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
>   * When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
>   * There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
> * Transactions page
>   
>   * For a given category, the list of transactions is presented, ordered by the most recent.
>   * At the top of the page the user could see the total amount for the category (the sum of all of the amounts of the transactions in that category).
>   * There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
>   * When the user clicks on the "Back" button (<), the user navigates to the home page.
> * "Add a new category" page
>   
>   * The user fills out a form to create a new category, indicating their name and icon (both mandatory).
>   * The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
>   * When the user clicks on the "Back" button (<), the user navigates to the home page.
> * "Add a new transaction" page
>   
>   * The user fills out a form to create a new transaction with the following:
>     * name (mandatory)
>     * amount (mandatory)
>     * categories (mandatory at least one)
>   * The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
>   * When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,161 @@
...
+<!-- SUPPORT -->
+
+## :star:️ Show your support <a name="support"></a>
+If you like this project you can fork it.
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## :pray: Acknowledgments <a name="acknowledgements"></a>
+I would like to thank... <a href="https://www.microverse.org/?gclid=CjwKCAiArY2fBhB9EiwAWqHK6s-2-x4d57Pghz47XT1BgsYuF81ZprM-k-IwzI0_L96nV0SQ93A8ExoCVnQQAvD_BwE" title="planet icons">Microvers for giving me this opportunity to build this project</a>
```
Comment (KDlamini) [inline README.md]: Kindly make sure that The README "Acknowledgements" section or similar gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: README.md [hunk 2]
```diff
@@ -1,2 +1,161 @@
...
+- It has user authentication.
+- It is well designed.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- PROJECT LINK-->
+
+## :link: Project Link <a name="project link"></a>
+- [Project link](https://github.com/Osoro254Alex/Budget-App)
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (KDlamini) [inline README.md]: Kindly make sure the application is deployed and that you have a live demo section that has a working link to the deployed URL.

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div class="container flex flex-col h-screen bg-slate-100 px-0">
+  <%= form_for(resource, as: resource_name, url: session_path(resource_name), method: :post) do |f| %>
+    <div class="flex justify-between items-center bg-blue py-4 px-2">
+      <%= link_to root_path do %>
+        <span class="text-white" style="font-size:24px;">&#10229;</span>
+      <% end %>
+      <span class="uppercase text-white">login</span>
+      <%= f.submit "Log in", class: "text-white p-1 font-semibold" %>
```
Comment (KDlamini) [inline app/views/devise/sessions/new.html.erb:8]: Well done nailing the design for the Login and Registration page. Impressive!!! 👏🏾 💯 🥳 

image

image

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<div class="container flex flex-col justify-between h-screen bg-slate-100 px-0">
+  <div class="relative basis-1 m-0">
+    <div class="flex justify-between items-center bg-blue py-4 px-4">
+       <%= render "shared/header" %>
```
Comment (KDlamini) [inline app/views/groups/index.html.erb:4]: Fantastic job nailing the design and functionalities for the categories and transaction page. Everything is on point 👌🏾🔥🔥🔥

✔️  The user can fill out a form to create a new category, indicating their name and icon.
✔️  The user fills out a form to create a new transaction with name, amount, and categories.

image

image

File: spec/controllers/expenses_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
```
Comment (KDlamini) [inline spec/controllers/expenses_controller_spec.rb:2]: Unit and Integration specs are well-written and sufficient covering all controllers, views, and models 🎆 🍾 

image

Well done! 💯

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Osoro254Alex 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working Budget app using Rails 7. 🎊 🥳 
Your project is really impressive. You have done a fantastic job!! 🥇
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞

✔️  Well documented README
✔️  There's user authentication, register and log in so that the data is private to them
✔️  Design Guidelines are followed
✔️  Model validations are present
✔️  Controllers, Migrations, Views, and Routes are set up correctly

## Video Presentation

✔️ The is a recorded presentation video presenting the features of the project.
✔️ The maximum length of the video is 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️ The video contains a demo of the project, highlighting all the project features.
✖️  The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Required Changes ♻️

- Kindly add a Pull Request title that is descriptive of this project like "Budget App - Ruby on Rails Capstone Project" for example.

**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @Osoro254Alex 👋 

### [OPTIONAL]
For Future Projects, Please take care of the following:

- For group capstone projects the correct flow for requesting reviews is Github flow which means dev merging to main.

- For solo capstone projects the flow for requesting reviews is Gitflow which means feature branch merging to dev.


## Things to Appreciate 
- [ ] Correct Flow ✅ 
- [ ] Clean Code 👍 
- [ ] No Linter Errors 🔥 
- [ ] Descriptive Pull Request and Summary 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Osoro254Alex** [comment]
@KDlamini  and @zainsadaqat Thanks

**Pull request merged**